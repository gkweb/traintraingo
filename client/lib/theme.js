import React, {Component} from 'react'

const themes = {
  light: {
    'name': 'light',
    'values': {
      'primaryBg': '#f7f7f7',
      'secondaryBg': '#ddd',
      'tertiaryBg': '#bbb',
      'inversePrimaryBg': '#f7f7f7',
      'primary': '#4c4c4c',
      'secondary': '#9a9a9a',
      'tertiary': '#f7f7f7',
      'inversePrimary': '#4c4c4c',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#4c4c4c',
      'logoSecondary': '#fff',
      'input': '#222',
      'inputBg': '#fff',
      'inputBorder': '#ccc',
      'inputIcon': 'rgb(76, 76, 76, .75)',
      'link': '#4495d4',
      'linkBg': '#f7f7f7',
      'linkHoverBg': '#fff'
    }
  },
  dark: {
    'name': 'dark',
    'values': {
      'primaryBg': '#444',
      'secondaryBg': '#333',
      'tertiaryBg': '#222',
      'inversePrimaryBg': '#fff',
      'primary': '#fff',
      'secondary': '#ddd',
      'tertiary': '#000',
      'inversePrimary': '#000',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#fff',
      'logoSecondary': '#fff',
      'input': '#fff',
      'inputBg': '#222',
      'inputBorder': '#000',
      'inputIcon': 'rgba(255, 255, 255, .25)',
      'link': '#fff',
      'linkBg': '#333',
      'linkHoverBg': '#222'
    }
  }
}

const ThemeManagementContext = React.createContext({
    activeTheme: themes.light.name,
    theme: themes.light // default value
  }
)

export {
  themes,
  ThemeManagementContext
}
