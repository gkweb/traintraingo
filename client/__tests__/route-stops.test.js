import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { themes } from './../lib/theme'
import RouteStops, { STOPPING_PATTERN_QUERY } from '../components/route-stops'
import 'jest-styled-components'
import { MockedProvider } from 'react-apollo/test-utils'
import renderer from 'react-test-renderer'
import waitForExpect from 'wait-for-expect'

// Helps with Link.prefetch mocking
jest.mock('next/link', () => {
  return ({ children }) => {
    return children
  }
})

const MockPattern = {
  pattern: {
    departures: [
      {
        stop_id: '1071',
        scheduled_departure_utc: '2019-05-16T08:34:00Z',
        estimated_departure_utc: '2019-05-16T08:33:40Z',
        route_id: 5,
        direction_name: 'Mernda',
        stop_name: 'Flinders Street',
        stop_suburb: 'Melbourne City',
      },
      {
        stop_id: '1181',
        scheduled_departure_utc: '2019-05-16T08:37:00Z',
        estimated_departure_utc: '2019-05-16T08:37:48Z',
        route_id: 5,
        direction_name: 'Mernda',
        stop_name: 'Southern Cross',
        stop_suburb: 'Melbourne City',
      },
      {
        stop_id: '1068',
        scheduled_departure_utc: '2019-05-16T08:39:00Z',
        estimated_departure_utc: '2019-05-16T08:39:48Z',
        route_id: 5,
        direction_name: 'Mernda',
        stop_name: 'Flagstaff',
        stop_suburb: 'Melbourne City',
      },
    ],
  },
}

// Stops Mock
const mocks = [
  {
    request: {
      query: STOPPING_PATTERN_QUERY,
      variables: {
        run_id: 949743,
      },
    },
    result: {
      data: MockPattern,
    },
  },
]

// Dom node mock

function createNodeMock() {
  // You can return anything from this function.
  // For example:
  return {
    scrollLeft: () => {
      // Do nothing
    },
  }
}

// Displays station names once clicked
// Displays Loading when empty data is provided

test('Displays station names once clicked', async () => {
  const c = renderer.create(
    <ThemeProvider theme={themes.light.values}>
      <MockedProvider
        mocks={mocks}
        variables={{ run_id: 949743 }}
        addTypename={false}
        removeTypename
      >
        <RouteStops runId={949743} stopId={1068} />
      </MockedProvider>
    </ThemeProvider>,
    { createNodeMock }
  )

  c.root.findByType('button').props.onClick()

  await waitForExpect(() => {
    expect(JSON.stringify(c.toJSON())).toMatch('Flagstaff')
    expect(JSON.stringify(c.toJSON())).toMatch('Southern Cross')
    expect(JSON.stringify(c.toJSON())).toMatch('Flinders Street')
  })
})

test('Displays loading text', async () => {
  const c = renderer.create(
    <ThemeProvider theme={themes.light.values}>
      <MockedProvider
        mocks={[]}
        variables={{ run_id: 949743 }}
        addTypename={false}
        removeTypename
      >
        <RouteStops runId={949743} stopId={1068} />
      </MockedProvider>
    </ThemeProvider>
  )

  c.root.findByType('button').props.onClick()

  console.log(c.root)

  await waitForExpect(() => {
    expect(JSON.stringify(c.toJSON())).toMatch('Loading...')
  })
})

test('Run stopping pattern Snapshot', () => {
  const c = shallow(<RouteStops runId={949743} stopId={1068} />, {
    theme: themes.light.values,
  })
  expect(c).toMatchSnapshot()
})
