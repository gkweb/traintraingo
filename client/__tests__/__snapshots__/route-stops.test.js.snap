// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run stopping pattern Snapshot 1`] = `
<Fragment>
  <styled.h4>
    Route stops
     
    <RefreshBtn
      onClick={[Function]}
    />
  </styled.h4>
</Fragment>
`;
